torch/_prims/__init__.py

```diff
@@ -253,16 +257,21 @@ def _elementwise_meta(
 
     assert len(args) > 0
 
-    utils.check_same_device(*args, allow_cpu_scalar_tensors=True)
-    utils.check_same_shape(*args, allow_cpu_scalar_tensors=True)
     utils.check_same_dtype(*args)
 
-    strides = utils.compute_elementwise_output_strides(*args)
+    args_ = list(args)
+    if args_with_different_dtypes is not None:
+        args_.extend(args_with_different_dtypes)
+
+    utils.check_same_device(*args_, allow_cpu_scalar_tensors=True)
+    utils.check_same_shape(*args_, allow_cpu_scalar_tensors=True)
+
+    strides = utils.compute_elementwise_output_strides(*args_)
 
     tensor = None
     scalar_tensor = None
     number = None
-    for arg in args:
+    for arg in args_:
```

Comment:
this would potentially set `tensor` to something with wrong dtype (from `args_with_different_types`)

Reply:
Great catch - fixed